internal/service/project/model_project.go

```diff
@@ -0,0 +1,154 @@
+			RoleNames: roleNames,
+		}
+	}
+	return teams
+}
+
+func NewTFLimitsDataSourceModel(ctx context.Context, dataFederationLimits []admin.DataFederationLimit) []*TfLimitModel {
+	limits := make([]*TfLimitModel, len(dataFederationLimits))
+
+	for i, dataFederationLimit := range dataFederationLimits {
+		limits[i] = &TfLimitModel{
+			Name:         types.StringValue(dataFederationLimit.Name),
+			Value:        types.Int64Value(dataFederationLimit.Value),
+			CurrentUsage: types.Int64PointerValue(dataFederationLimit.CurrentUsage),
+			DefaultLimit: types.Int64PointerValue(dataFederationLimit.DefaultLimit),
+			MaximumLimit: types.Int64PointerValue(dataFederationLimit.MaximumLimit),
+		}
+	}
+
+	return limits
+}
+
+func NewTFProjectResourceModel(ctx context.Context, projectRes *admin.Group,
+	teams *admin.PaginatedTeamRole, projectSettings *admin.GroupSettings, limits []admin.DataFederationLimit) *TfProjectRSModel {
+	projectPlan := TfProjectRSModel{
+		ID:                        types.StringValue(projectRes.GetId()),
+		Name:                      types.StringValue(projectRes.Name),
+		OrgID:                     types.StringValue(projectRes.OrgId),
+		ClusterCount:              types.Int64Value(projectRes.ClusterCount),
+		Created:                   types.StringValue(conversion.TimeToString(projectRes.Created)),
+		WithDefaultAlertsSettings: types.BoolPointerValue(projectRes.WithDefaultAlertsSettings),
+		Teams:                     newTFTeamsResourceModel(ctx, teams),
+		Limits:                    newTFLimitsResourceModel(ctx, limits),
+	}
+
+	if projectSettings != nil {
+		projectPlan.IsCollectDatabaseSpecificsStatisticsEnabled = types.BoolValue(*projectSettings.IsCollectDatabaseSpecificsStatisticsEnabled)
+		projectPlan.IsDataExplorerEnabled = types.BoolValue(*projectSettings.IsDataExplorerEnabled)
+		projectPlan.IsExtendedStorageSizesEnabled = types.BoolValue(*projectSettings.IsExtendedStorageSizesEnabled)
+		projectPlan.IsPerformanceAdvisorEnabled = types.BoolValue(*projectSettings.IsPerformanceAdvisorEnabled)
+		projectPlan.IsRealtimePerformancePanelEnabled = types.BoolValue(*projectSettings.IsRealtimePerformancePanelEnabled)
+		projectPlan.IsSchemaAdvisorEnabled = types.BoolValue(*projectSettings.IsSchemaAdvisorEnabled)
+	}
+
+	return &projectPlan
+}
+
+func newTFLimitsResourceModel(ctx context.Context, dataFederationLimits []admin.DataFederationLimit) types.Set {
+	limits := make([]TfLimitModel, len(dataFederationLimits))
+
+	for i, dataFederationLimit := range dataFederationLimits {
+		limits[i] = TfLimitModel{
+			Name:         types.StringValue(dataFederationLimit.Name),
+			Value:        types.Int64Value(dataFederationLimit.Value),
+			CurrentUsage: types.Int64PointerValue(dataFederationLimit.CurrentUsage),
+			DefaultLimit: types.Int64PointerValue(dataFederationLimit.DefaultLimit),
+			MaximumLimit: types.Int64PointerValue(dataFederationLimit.MaximumLimit),
+		}
+	}
+
+	s, _ := types.SetValueFrom(ctx, TfLimitObjectType, limits)
```

Comment:
Agreed that we probably should check for errors. I didn't change it because this method already existed in `internal/service/project/resource_project.go` so I have avoided to change behaviour.  I only moved it